### ISSUE TITLE
PMT presence is causing raw stream detection to fail.

### ISSUE THREAD

**Issue by enternoescape** [google/sagetv#197]
A valid PAT and all of the PMT's are present, but none of the PMT's reference the actual streams available in the transport stream. Admittedly, this makes the file not work with a lot more players than just SageTV and once again, the provider doing things badly is Comcast. This happens for at least one user on two different channels consistently. Below is a video clip with the issue.

https://drive.google.com/open?id=0B4AD2tiek6XiUjhlaUw0dExNNlE

I propose we could address this situation by removing the presence of the PMT packet as a reason we will not eventually just use all of the streams we have found after a specific amount of data has been processed. As a fail-safe, I would make the presence of valid PMT packets raise the threshold for a naked stream significantly so it's less likely that we accidentally end up with a naked stream when it just happens to be a really high bit-rate transport stream.

**Comment by CraziFuzzy** [comment]
Is this an issue that should also be brought up with silicondust? Seems like it's packet filtering should be excluding the pmts that aren't for the selected program.

**Comment by enternoescape** [comment]
I suppose we cannot rule out a bug in the Prime, so I agree that should be investigated too, if not first.

I don't know exactly what virtual channel information DCT's get from the CableCARD, but I suspect they are providing the frequency and program number based on what you can get from the InfiniTV cards, so it's a little odd that the a PMT referencing the correct PID's is missing. We know in #152 the real issue is that the PMT packet isn't being decrypted, so I can understand in that case why it might still know what the PID's should be, but here I'm fairly confused.

**Comment by qianzhang5** [comment]
If there are one more video stream in a TV channel and a video has no
correct PMT to be associated with,  you can't rebuild PMT packet as well. I
think there is other reason cause trouble.

On Sat, Oct 15, 2016 at 9:10 AM, Christopher Piper <notifications@github.com

**Comment by qianzhang5** [comment]
If silicondust handles a streamer in front end, I think it's their bug. but
in this case you may rebuild a PMT for a video, because there is only one
video data stream a channel.

**Comment by CraziFuzzy** [comment]
Well, we have seen that the Prime does create a new PAT, with JUST the selected program in it, right (and that is the first packet sent).  So later, they are sending PMT's that are associated with programs NOT in their recreated PAT - that alone seems like an issue with how THEY are remuxing it.  To me, if they are going to filter to a specific vchannel (program), than they should be filtering completely - that means no packets should be sent downstream that aren't:
PAT with a single program (the selected one)
PMT for that program
video and audio packets referenced in that PMT.

Any other packets are superfluous, and confusing.

**Comment by CraziFuzzy** [comment]
of course, we should also see if they can implement a way to decrypt and grab the entire tuned stream, with full untouched packets as well.

**Comment by enternoescape** [comment]
Since I personally don't have this issue, it's up to the user to report it and see what Silicondust will do.

**Comment by enternoescape** [comment]
I was mistaken. This is a Ceton InfiniTV device, so if this is a bug, it's not going to be fixed.
